pkg/vendir/config/config.go

```diff
@@ -278,3 +280,12 @@ func (c Config) checkOverlappingPaths() error {
 
 	return nil
 }
+
+func (c *Config) cleanPaths() {
```

Comment:
Since we are change the configuration directly will this have any impact on the lazy feature?

Reply:
I'm not sure I understand the question.

Paths are clean up right after reading the config, so after that any operation uses so to say "canonical" path, including the lock file and lazy functionality.

Reply:
But if the lock file already exists, the path might be different correct?

Reply:
But if the lock file already exists, the path might be different correct?

Reply:
Yes, the current vendir behavior is to copy paths from the config to the lock file literally. With the cleanup logic, paths can differ between the config and the lock file. It is basically the same as changing `vendor/` to `vendor` in the config and run sync. The lock file will be updated with the new path.

So, yes, if this is merged, the next sync with the new version of vendir will invalidate the "lazy cache" in some cases. But I don't think that can be an issue.